src/main/java/net/dries007/tfc/common/blockentities/PowderkegBlockEntity.java

```diff
@@ -0,0 +1,268 @@
+/*
+ * Licensed under the EUPL, Version 1.2.
+ * You may obtain a copy of the Licence at:
+ * https://joinup.ec.europa.eu/collection/eupl/eupl-text-eupl-12
+ */
+
+package net.dries007.tfc.common.blockentities;
+
+import net.dries007.tfc.common.blocks.devices.PowderkegBlock;
+import net.dries007.tfc.common.capabilities.*;
+import net.dries007.tfc.common.container.PowderkegContainer;
+import net.dries007.tfc.common.recipes.inventory.EmptyInventory;
+import net.dries007.tfc.util.Helpers;
+import net.dries007.tfc.util.PowderKegExplosion;
+import net.dries007.tfc.util.calendar.Calendars;
+import net.minecraft.core.BlockPos;
+import net.minecraft.core.Direction;
+import net.minecraft.nbt.CompoundTag;
+import net.minecraft.nbt.ListTag;
+import net.minecraft.nbt.Tag;
+import net.minecraft.network.chat.Component;
+import net.minecraft.network.chat.TranslatableComponent;
+import net.minecraft.sounds.SoundEvents;
+import net.minecraft.sounds.SoundSource;
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.entity.player.Inventory;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.inventory.AbstractContainerMenu;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.item.Items;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.level.block.Blocks;
+import net.minecraft.world.level.block.state.BlockState;
+import net.minecraftforge.common.capabilities.Capability;
+import net.minecraftforge.common.util.INBTSerializable;
+import net.minecraftforge.common.util.LazyOptional;
+import net.minecraftforge.items.CapabilityItemHandler;
+import net.minecraftforge.items.IItemHandlerModifiable;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class PowderkegBlockEntity extends TickableInventoryBlockEntity<PowderkegBlockEntity.PowderkegInventory>
+{
+
+    private int fuse = -1;
+    private static final int SLOTS = 12;
+    public static final int SLOT_INPUT_START = 0;
+    public static final int SLOT_INPUT_END = 11;
+    private boolean isLit = false;
+    private Entity igniter;
```

Comment:
In our code style xml there's a rule defined which groups things in the order static (in order of decreasing visibility), then non static (in order of decreasing visibility). In addition since `Entity` is not initialized in the constructor, and due to `@FieldsAreNonNullByDefault`, it should be annotated as `@Nullable`
```suggestion
    public static final int SLOT_INPUT_START = 0;
    public static final int SLOT_INPUT_END = 11;

    private static final int SLOTS = 12;

    private int fuse = -1;
    private boolean isLit = false;
    private @Nullable Entity igniter;
```